feat : define cocountable and cocardinal filters

JADekker opened the pull request:
Define cocountable and cocardinal filters.

Any feedback on the way these are phrased and written is more than welcome! 

---
<!-- The text above the `---` will become the commit message when your
PR is merged. Please leave a blank newline before the `---`, otherwise
GitHub will format the text above it as a title.

To indicate co-authors, include lines at the bottom of the commit message
(that is, before the `---`) using the following format:

Co-authored-by: Author Name <author@email.com>

Any other comments you want to keep out of the PR commit should go
below the `---`, and placed outside this HTML comment, or else they
will be invisible to reviewers.

If this PR depends on other PRs, please list them below this comment,
using the following format:
- [ ] depends on: #abc [optional extra text]
- [ ] depends on: #xyz [optional extra text]
-->

[![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/from-referrer/)

JADekker committed:
feat : add cocardinal filters

JADekker committed:
Merge branch 'master' into JADekker_topology_additions

JADekker committed:
feat : add cocardinal filter

JADekker committed:
Merge branch 'master' into JADekker_topology_additions

JADekker committed:
feat : put results in correct files

JADekker committed:
fix : add new file

urkud commented on Mathlib/Topology/Defs/Filter.lean:
```diff
@@ -211,7 +211,6 @@ def MapClusterPt {ι : Type*} (x : X) (F : Filter ι) (u : ι → X) : Prop :=
 See also `ClusterPt`. -/
 def AccPt (x : X) (F : Filter X) : Prop :=
   NeBot (𝓝[≠] x ⊓ F)
-#align acc_pt AccPt
```
Why do you drop an `#align`?

JADekker replied:
I have no clue, I must've accidentally done this and missed that this change was made, I'll revert that!

JADekker committed:
Update Cocardinal.lean

JADekker committed:
feat : change definition of ofCardinalInter

Require a strict inequality for
- ofCardinalInter
- ofCardinalUnion
- cocardinal

JADekker committed:
fix : lint issue

urkud commented on Mathlib/Order/Filter/Cocardinal.lean:
```diff
@@ -0,0 +1,70 @@
+  exact hUnion (compl '' S) (lt_of_le_of_lt Cardinal.mk_image_le hSc) (ball_image_iff.2 hSp)
+
+-- Of course, this would generalise to CardinalInterFilter under a suitable definition for these.
+instance countableInter_ofCardinalnter {c : Cardinal} (hc : Cardinal.aleph0 < c) (l : Set (Set α))
```
This can't be an `instance` because Lean can't guess `hc : aleph0 < c` from available `hc.le : aleph0 ≤ c`.

JADekker replied:
But you only get a `CountableInterFilter` from this construction if `aleph0 < c`. Otherwise, we only have stability under finite intersections.

JADekker replied:
Perhaps we may want `aleph0 < c` everywhere anyway, so that this explicitly does not nest the case for finite intersections?

JADekker replied:
I've redefined some things, I think the problem should be gone now.

JADekker committed:
Merge branch 'master' into JADekker_topology_additions

JADekker commented:
#10531 defines `CardinalInterFilter`, I'll make sure that both PRs use similar design choices. We should be able to conclude that `cocardinal` filters (cardinality `c`) are `CardinalInterFilter`s for all cardinalities strictly lower than `c`.

JADekker committed:
Merge branch 'master' into JADekker_topology_additions

JADekker committed:
feat : fix definition of cocardinal

JADekker committed:
fix : lint

JADekker commented:
I've removed the 'Awaiting-Review' label for now, as I'll first build up the theory of `CardinalInterFilters` more, these were introduced in #10531

JADekker committed:
Merge branch 'master' into JADekker_topology_additions

JADekker committed:
minor change

JADekker commented:
This PR has been closed in favour of a newer PR, which makes better use of the recently added `CardinalInterFilter` file.
